### DISCUSSION TITLE
404 error  for /public/static just uploaded files only in production mode

### DISCUSSION THREAD

**Discussion by haw0k** [vercel/next.js#17201] [Help] [answered]
My project is based on NextJS 9 custom server. Custom server is Express v4.

I upload files through the multipart form to the server into /public/static/ folder by multer. 

In production mode I always receive response with 404 code for GET-requests to just uploaded files. 
After server restart I receive response with 200 code for all previously uploaded files and 404 code for new uploads only.

This issue does not exist in developer mode.

Problem is not present only if
server.js contained:
`const nextApp = nextjs({ dev }); `
and server launch command is:
`$ nodemon server.js`

[github project ](https://github.com/haw0k/ds-collections)

**Comment by laugharn** [comment, marked as answer]
https://github.com/vercel/next.js/discussions/16417 the public directory isn't an actual directory, it is a collection of mapped routes created at build time.

**Reply by haw0k** [reply]
Unexpected solution (like a workaround).

Thank you!

**Comment by forrestwilkins** [comment]
If anyone else's using nginx, the following solved it for me.

```
    location ^~ /uploads/ {
        alias /home/node/app_name/public/uploads/;
        sendfile on;
    }
```

**Reply by Umar-Farooq-Shafi** [reply]
Save my day. thanks

**Reply by hugoafj** [reply]
save my week
